Zend/tests/attributes/013_class_scope.phpt

```diff
@@ -0,0 +1,105 @@
+--TEST--
+Attributes make use of class scope.
+--FILE--
+<?php
+
+<<A1(self::class, self::FOO)>>
+class C1
+{
+	<<A1(self::class, self::FOO)>>
+	private const FOO = 'foo';
+
+	<<A1(self::class, self::FOO)>>
+	public $a;
+
+	<<A1(self::class, self::FOO)>>
+	public function bar(<<A1(self::class, self::FOO)>> $p) { }
+}
+
+$ref = new \ReflectionClass(C1::class);
+print_r($ref->getAttributes()[0]->getArguments());
+print_r($ref->getReflectionConstant('FOO')->getAttributes()[0]->getArguments());
+print_r($ref->getProperty('a')->getAttributes()[0]->getArguments());
+print_r($ref->getMethod('bar')->getAttributes()[0]->getArguments());
+print_r($ref->getMethod('bar')->getParameters()[0]->getAttributes()[0]->getArguments());
+
+echo "\n";
+
+trait T1
+{
+	<<A1(self::class, self::FOO)>>
+	public function foo() { }
+}
+
+class C2
+{
+	use T1;
+	
+	private const FOO = 'bar';
+}
+
+$ref = new \ReflectionClass(C2::class);
+print_r($ref->getMethod('foo')->getAttributes()[0]->getArguments());
```

Comment:
Please also check the same on T1 directly (which should give a non-existing class constant exception).